trio/_core/_multierror.py

```diff
@@ -87,7 +87,9 @@ def filter_tree(
             new_exceptions = []
             changed = False
             for child_exc in exc.exceptions:
-                new_child_exc = filter_tree(child_exc, preserved)
+                new_child_exc = filter_tree(  # noqa: F821  # Deleted in local scope below, causes ruff to think it's not defined
```

Comment:
Do we have an upstream issue to point to? I know we talked about it a bit in the PR comments but a link is always nice, especially if it's solely about that issue :D.

Reply:
There does not appear to be an upstream issue with not very detailed searching.

Sorry to bother you, but @charliermarsh is this a bug or intended and will not be changing?

Reply:
No worries -- I think this can be considered a bug. I created an issue to track it: https://github.com/astral-sh/ruff/issues/7733.

Reply:
Added upstream issue comment in c53f3d1c